src/problem-matcher.js

```diff
@@ -0,0 +1,42 @@
+import { info } from '@actions/core';
+import { writeFileSync } from 'fs';
+
+const json = {
+    problemMatcher: [
+        {
+            owner: 'packsquash-error',
+            severity: 'error',
+            pattern: [
+                {
+                    regexp: '^[!❌] (?!.*Invalid stick parity bit)(.+)$',
+                    message: 1
+                }
+            ]
+        },
+        {
+            owner: 'packsquash-warning',
+            severity: 'warning',
+            pattern: [
+                {
+                    regexp: '^[*⚡] (.+)$',
+                    message: 1
+                }
+            ]
+        }
+    ]
+};
+
+/**
+ * @param {string} path
+ */
+export function addProblemMatcher(path) {
+    writeFileSync(path, JSON.stringify(json));
+    info(`##[add-matcher]${path}`);
+}
+
+/**
+ * @param {string} path
+ */
+export function removeProblemMatcher(path) {
+    info(`##[remove-matcher]${path}`);
```

Comment:
For some reason, neither `##[add-matcher]` or `##[remove-matcher]` are documented officially by GitHub, even though they are used by some actions they've made. To be on the safe side, I'd prefer to use the [better-documented commands](https://github.com/actions/toolkit/blob/1cc56db0ff126f4d65aeb83798852e02a2c180c3/docs/commands.md#problem-matchers) instead, like the shell script was doing.

Reply:
Fix by e47aca6b7bf6bb2db11ac2b7bd204ba39bdcbddd

As an aside, I found the command from the official setup action. This command is used in all the examples in [the problem-matcher.json documentation](https://github.com/actions/toolkit/blob/1cc56db0ff126f4d65aeb83798852e02a2c180c3/docs/problem-matchers.md). For example:
- https://github.com/actions/setup-node/blob/bacd6b4b3ac3127b28a1e1920c23bf1c2cadbb85/src/main.ts#L54
- https://github.com/actions/setup-dotnet/blob/f078482971ee996a7da79a0b59e838f5049eb263/src/setup-dotnet.ts#L62

There are two commands available, and, strangely, the commands which official actions use and the documented commands are different.

Reply:
Yes, I can see why you chose the alternative 👍

However, in my opinion, it's more difficult to keep track of features that were only ever "documented" in examples. By choosing the commands that are explicitly mentioned in the documentation, we buy ourselves a "safety net", as it is easier to `grep` proper documentation or changelogs for any changes compared to having to look at someone's code. The documentation also serves as a "contract" between GitHub and developers, as any changes like this to it is obviously breaking, while changes to examples generally are not.

Reply:
> My only concern is whether the ::remove-matcher command works by path too, as the documentation only explicitly mentions that it can remove matchers by owner. I think we should add a new test workflow, optimizing an invalid JSON file or something like that, to check whether the problem matchers work as intended.

<img width="967" alt="image" src="https://user-images.githubusercontent.com/34268371/168099751-56981ece-16ae-4210-8844-9b4bc0272943.png">

remove-matcher is also working fine 😉

Reply:
Perfect!!